src/native/corehost/bundle/file_entry.cpp

```diff
@@ -19,13 +19,16 @@ file_entry_t file_entry_t::read(reader_t &reader, uint32_t bundle_major_version,
     // First read the fixed-sized portion of file-entry
     file_entry_fixed_t fixed_data;
 
-    fixed_data.offset = *(int64_t*)reader.read_direct(sizeof(int64_t));
-    fixed_data.size   = *(int64_t*)reader.read_direct(sizeof(int64_t));
+    // NB: the file data is potentially unaligned, thus we use "read" to fetch 64bit values
+    reader.read(&fixed_data.offset, sizeof(int64_t));
+    reader.read(&fixed_data.size, sizeof(int64_t));
 
     // compressedSize is present only in v6+ headers
-    fixed_data.compressedSize = bundle_major_version >= 6 ?
-                        *(int64_t*)reader.read_direct(sizeof(int64_t)) :
-                        0;
+    fixed_data.compressedSize = 0;
+    if (bundle_major_version >= 6)
+    {
+        reader.read(&fixed_data.compressedSize, sizeof(int64_t));
+    }
 
     fixed_data.type   = *(file_type_t*)reader.read_direct(sizeof(file_type_t));
```

Comment:
What is the size of `file_type_t` here and how is it guaranteed to be aligned? or do reads of its size not need aligning?

Reply:
It is an enum with base uint8_t, defined here:
https://github.com/dotnet/runtime/blob/7fc4b6ae1e70a53aaa63b748e292b4cb8fe82c2b/src/native/corehost/bundle/file_type.h#L19